exchange/binance_future.go

```diff
@@ -442,56 +445,23 @@ func (b *BinanceFuture) Position(pair string) (asset, quote float64, err error)
 	return assetBalance.Free + assetBalance.Lock, quoteBalance.Free + quoteBalance.Lock, nil
 }
 
-func (b *BinanceFuture) CandlesSubscription(ctx context.Context, pair, period string) (chan model.Candle, chan error) {
+func (b *BinanceFuture) CandlesSubscriptionCombined(ctx context.Context,
+	pairs map[string]string) (chan model.Candle, chan error) {
 	ccandle := make(chan model.Candle)
 	cerr := make(chan error)
 	ha := model.NewHeikinAshi()
 
-	go func() {
-		ba := &backoff.Backoff{
-			Min: 100 * time.Millisecond,
-			Max: 1 * time.Second,
-		}
-
-		for {
-			done, _, err := futures.WsKlineServe(pair, period, func(event *futures.WsKlineEvent) {
-				ba.Reset()
-				candle := FutureCandleFromWsKline(pair, event.Kline)
-
-				if candle.Complete && b.HeikinAshi {
-					candle = candle.ToHeikinAshi(ha)
-				}
-
-				if candle.Complete {
-					// fetch aditional data if needed
-					for _, fetcher := range b.MetadataFetchers {
-						key, value := fetcher(pair, candle.Time)
-						candle.Metadata[key] = value
-					}
-				}
+	go b.candleSubscription(ctx, ha, ccandle, cerr, "", "", pairs, true)
 
-				ccandle <- candle
+	return ccandle, cerr
+}
 
-			}, func(err error) {
-				cerr <- err
-			})
-			if err != nil {
-				cerr <- err
-				close(cerr)
-				close(ccandle)
-				return
-			}
+func (b *BinanceFuture) CandlesSubscription(ctx context.Context, pair, period string) (chan model.Candle, chan error) {
+	ccandle := make(chan model.Candle)
+	cerr := make(chan error)
+	ha := model.NewHeikinAshi()
 
-			select {
-			case <-ctx.Done():
-				close(cerr)
-				close(ccandle)
-				return
-			case <-done:
-				time.Sleep(ba.Duration())
-			}
-		}
-	}()
+	go b.candleSubscription(ctx, ha, ccandle, cerr, pair, period, nil, false)
```

Comment:
why `pairs` parameter is nil here?

Reply:
Because method `candleSubscription` handles both cases: for single pair-period (CandlesSubscription) and for multiple pairs (CandlesSubscriptionCombined)